### ISSUE TITLE
Installation on Windows 8.1 failed

### ISSUE THREAD

**Issue by jankogaga** [NicolasHug/Surprise#28]
I have an issue of Surprise installation on Windows 8.1. 

When running
 
pip install scikit-surprise

from command line, I get the following output:

```
C:\Users\Drugi>pip install scikit-surprise
Collecting scikit-surprise
  Using cached scikit-surprise-1.0.3.tar.gz
Requirement already satisfied: numpy>=1.11.2 in c:\programdata\anaconda3\lib\si
e-packages (from scikit-surprise)
Requirement already satisfied: six>=1.10.0 in c:\programdata\anaconda3\lib\site
packages (from scikit-surprise)
Building wheels for collected packages: scikit-surprise
  Running setup.py bdist_wheel for scikit-surprise ... error
  Complete output from command C:\ProgramData\Anaconda3\python.exe -u -c "impor
 setuptools, tokenize;__file__='C:\\Users\\Drugi\\AppData\\Local\\Temp\\pip-bui
d-qbu2qjqt\\scikit-surprise\\setup.py';f=getattr(tokenize, 'open', open)(__file
_);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, '
xec'))" bdist_wheel -d C:\Users\Drugi\AppData\Local\Temp\tmp_h27f0pzpip-wheel-
-python-tag cp36:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.6
  creating build\lib.win-amd64-3.6\surprise
  copying surprise\accuracy.py -> build\lib.win-amd64-3.6\surprise
  copying surprise\dataset.py -> build\lib.win-amd64-3.6\surprise
  copying surprise\dump.py -> build\lib.win-amd64-3.6\surprise
  copying surprise\evaluate.py -> build\lib.win-amd64-3.6\surprise
  copying surprise\__init__.py -> build\lib.win-amd64-3.6\surprise
  copying surprise\__main__.py -> build\lib.win-amd64-3.6\surprise
  creating build\lib.win-amd64-3.6\surprise\prediction_algorithms
  copying surprise\prediction_algorithms\algo_base.py -> build\lib.win-amd64-3.
\surprise\prediction_algorithms
  copying surprise\prediction_algorithms\baseline_only.py -> build\lib.win-amd6
-3.6\surprise\prediction_algorithms
  copying surprise\prediction_algorithms\knns.py -> build\lib.win-amd64-3.6\sur
rise\prediction_algorithms
  copying surprise\prediction_algorithms\predictions.py -> build\lib.win-amd64-
.6\surprise\prediction_algorithms
  copying surprise\prediction_algorithms\random_pred.py -> build\lib.win-amd64-
.6\surprise\prediction_algorithms
  copying surprise\prediction_algorithms\__init__.py -> build\lib.win-amd64-3.6
surprise\prediction_algorithms
  running egg_info
  writing scikit_surprise.egg-info\PKG-INFO
  writing dependency_links to scikit_surprise.egg-info\dependency_links.txt
  writing entry points to scikit_surprise.egg-info\entry_points.txt
  writing requirements to scikit_surprise.egg-info\requires.txt
  writing top-level names to scikit_surprise.egg-info\top_level.txt
  warning: manifest_maker: standard file '-c' not found

  reading manifest file 'scikit_surprise.egg-info\SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  writing manifest file 'scikit_surprise.egg-info\SOURCES.txt'
  copying surprise\similarities.c -> build\lib.win-amd64-3.6\surprise
  copying surprise\similarities.pyx -> build\lib.win-amd64-3.6\surprise
  copying surprise\prediction_algorithms\co_clustering.c -> build\lib.win-amd64
3.6\surprise\prediction_algorithms
  copying surprise\prediction_algorithms\matrix_factorization.c -> build\lib.wi
-amd64-3.6\surprise\prediction_algorithms
  copying surprise\prediction_algorithms\optimize_baselines.c -> build\lib.win-
md64-3.6\surprise\prediction_algorithms
  copying surprise\prediction_algorithms\slope_one.c -> build\lib.win-amd64-3.6
surprise\prediction_algorithms
  copying surprise\prediction_algorithms\co_clustering.pyx -> build\lib.win-amd
4-3.6\surprise\prediction_algorithms
  copying surprise\prediction_algorithms\matrix_factorization.pyx -> build\lib.
in-amd64-3.6\surprise\prediction_algorithms
  copying surprise\prediction_algorithms\optimize_baselines.pyx -> build\lib.wi
-amd64-3.6\surprise\prediction_algorithms
  copying surprise\prediction_algorithms\slope_one.pyx -> build\lib.win-amd64-3
6\surprise\prediction_algorithms
  running build_ext
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\Drugi\AppData\Local\Temp\pip-build-qbu2qjqt\scikit-surprise\
etup.py", line 101, in <module>
      ['surprise = surprise.__main__:main']},
    File "C:\ProgramData\Anaconda3\lib\distutils\core.py", line 148, in setup
      dist.run_commands()
    File "C:\ProgramData\Anaconda3\lib\distutils\dist.py", line 955, in run_com
ands
      self.run_command(cmd)
    File "C:\ProgramData\Anaconda3\lib\distutils\dist.py", line 974, in run_com
and
      cmd_obj.run()
    File "C:\ProgramData\Anaconda3\lib\site-packages\wheel\bdist_wheel.py", lin
 179, in run
      self.run_command('build')
    File "C:\ProgramData\Anaconda3\lib\distutils\cmd.py", line 313, in run_comm
nd
      self.distribution.run_command(command)
    File "C:\ProgramData\Anaconda3\lib\distutils\dist.py", line 974, in run_com
and
      cmd_obj.run()
    File "C:\ProgramData\Anaconda3\lib\distutils\command\build.py", line 135, i
 run
      self.run_command(cmd_name)
    File "C:\ProgramData\Anaconda3\lib\distutils\cmd.py", line 313, in run_comm
nd
      self.distribution.run_command(command)
    File "C:\ProgramData\Anaconda3\lib\distutils\dist.py", line 974, in run_com
and
      cmd_obj.run()
    File "C:\ProgramData\Anaconda3\lib\site-packages\Cython\Distutils\old_build
ext.py", line 185, in run
      _build_ext.build_ext.run(self)
    File "C:\ProgramData\Anaconda3\lib\distutils\command\build_ext.py", line 30
, in run
      force=self.force)
    File "C:\ProgramData\Anaconda3\lib\distutils\ccompiler.py", line 1031, in n
w_compiler
      return klass(None, dry_run, force)
    File "C:\ProgramData\Anaconda3\lib\distutils\cygwinccompiler.py", line 282,
in __init__
      CygwinCCompiler.__init__ (self, verbose, dry_run, force)
    File "C:\ProgramData\Anaconda3\lib\distutils\cygwinccompiler.py", line 126,
in __init__
      if self.ld_version >= "2.10.90":
  TypeError: '>=' not supported between instances of 'NoneType' and 'str'

  ----------------------------------------
  Failed building wheel for scikit-surprise
  Running setup.py clean for scikit-surprise
Failed to build scikit-surprise
Installing collected packages: scikit-surprise
  Running setup.py install for scikit-surprise ... error
    Complete output from command C:\ProgramData\Anaconda3\python.exe -u -c "imp
rt setuptools, tokenize;__file__='C:\\Users\\Drugi\\AppData\\Local\\Temp\\pip-b
ild-qbu2qjqt\\scikit-surprise\\setup.py';f=getattr(tokenize, 'open', open)(__fi
e__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__,
'exec'))" install --record C:\Users\Drugi\AppData\Local\Temp\pip-l4ooc5n4-recor
\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.6
    creating build\lib.win-amd64-3.6\surprise
    copying surprise\accuracy.py -> build\lib.win-amd64-3.6\surprise
    copying surprise\dataset.py -> build\lib.win-amd64-3.6\surprise
    copying surprise\dump.py -> build\lib.win-amd64-3.6\surprise
    copying surprise\evaluate.py -> build\lib.win-amd64-3.6\surprise
    copying surprise\__init__.py -> build\lib.win-amd64-3.6\surprise
    copying surprise\__main__.py -> build\lib.win-amd64-3.6\surprise
    creating build\lib.win-amd64-3.6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\algo_base.py -> build\lib.win-amd64-
.6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\baseline_only.py -> build\lib.win-am
64-3.6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\knns.py -> build\lib.win-amd64-3.6\s
rprise\prediction_algorithms
    copying surprise\prediction_algorithms\predictions.py -> build\lib.win-amd6
-3.6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\random_pred.py -> build\lib.win-amd6
-3.6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\__init__.py -> build\lib.win-amd64-3
6\surprise\prediction_algorithms
    running egg_info
    writing scikit_surprise.egg-info\PKG-INFO
    writing dependency_links to scikit_surprise.egg-info\dependency_links.txt
    writing entry points to scikit_surprise.egg-info\entry_points.txt
    writing requirements to scikit_surprise.egg-info\requires.txt
    writing top-level names to scikit_surprise.egg-info\top_level.txt
    warning: manifest_maker: standard file '-c' not found

    reading manifest file 'scikit_surprise.egg-info\SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'scikit_surprise.egg-info\SOURCES.txt'
    copying surprise\similarities.c -> build\lib.win-amd64-3.6\surprise
    copying surprise\similarities.pyx -> build\lib.win-amd64-3.6\surprise
    copying surprise\prediction_algorithms\co_clustering.c -> build\lib.win-amd
4-3.6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\matrix_factorization.c -> build\lib.
in-amd64-3.6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\optimize_baselines.c -> build\lib.wi
-amd64-3.6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\slope_one.c -> build\lib.win-amd64-3
6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\co_clustering.pyx -> build\lib.win-a
d64-3.6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\matrix_factorization.pyx -> build\li
.win-amd64-3.6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\optimize_baselines.pyx -> build\lib.
in-amd64-3.6\surprise\prediction_algorithms
    copying surprise\prediction_algorithms\slope_one.pyx -> build\lib.win-amd64
3.6\surprise\prediction_algorithms
    running build_ext
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\Drugi\AppData\Local\Temp\pip-build-qbu2qjqt\scikit-surpris
\setup.py", line 101, in <module>
        ['surprise = surprise.__main__:main']},
      File "C:\ProgramData\Anaconda3\lib\distutils\core.py", line 148, in setup
        dist.run_commands()
      File "C:\ProgramData\Anaconda3\lib\distutils\dist.py", line 955, in run_c
mmands
        self.run_command(cmd)
      File "C:\ProgramData\Anaconda3\lib\distutils\dist.py", line 974, in run_c
mmand
        cmd_obj.run()
      File "C:\ProgramData\Anaconda3\lib\site-packages\setuptools-27.2.0-py3.6.
gg\setuptools\command\install.py", line 61, in run
      File "C:\ProgramData\Anaconda3\lib\distutils\command\install.py", line 54
, in run
        self.run_command('build')
      File "C:\ProgramData\Anaconda3\lib\distutils\cmd.py", line 313, in run_co
mand
        self.distribution.run_command(command)
      File "C:\ProgramData\Anaconda3\lib\distutils\dist.py", line 974, in run_c
mmand
        cmd_obj.run()
      File "C:\ProgramData\Anaconda3\lib\distutils\command\build.py", line 135,
in run
        self.run_command(cmd_name)
      File "C:\ProgramData\Anaconda3\lib\distutils\cmd.py", line 313, in run_co
mand
        self.distribution.run_command(command)
      File "C:\ProgramData\Anaconda3\lib\distutils\dist.py", line 974, in run_c
mmand
        cmd_obj.run()
      File "C:\ProgramData\Anaconda3\lib\site-packages\Cython\Distutils\old_bui
d_ext.py", line 185, in run
        _build_ext.build_ext.run(self)
      File "C:\ProgramData\Anaconda3\lib\distutils\command\build_ext.py", line
08, in run
        force=self.force)
      File "C:\ProgramData\Anaconda3\lib\distutils\ccompiler.py", line 1031, in
new_compiler
        return klass(None, dry_run, force)
      File "C:\ProgramData\Anaconda3\lib\distutils\cygwinccompiler.py", line 28
, in __init__
        CygwinCCompiler.__init__ (self, verbose, dry_run, force)
      File "C:\ProgramData\Anaconda3\lib\distutils\cygwinccompiler.py", line 12
, in __init__
        if self.ld_version >= "2.10.90":
    TypeError: '>=' not supported between instances of 'NoneType' and 'str'

    ----------------------------------------
Command "C:\ProgramData\Anaconda3\python.exe -u -c "import setuptools, tokenize
__file__='C:\\Users\\Drugi\\AppData\\Local\\Temp\\pip-build-qbu2qjqt\\scikit-su
prise\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().repl
ce('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --rec
rd C:\Users\Drugi\AppData\Local\Temp\pip-l4ooc5n4-record\install-record.txt --s
ngle-version-externally-managed --compile" failed with error code 1 in C:\Users
Drugi\AppData\Local\Temp\pip-build-qbu2qjqt\scikit-surprise\
```

Could you someone help me?

Regards,
Dragan

**Comment by NicolasHug** [comment]
Do you have C compiler on your machine? It looks like you don't even have a linker (hence the error with `ld_version` which is `None`).

I would suggest that you install visual studio (with the compiler). This is probably overkill but it will likely fix your issue, and I'm really not familiar enough with Windows to point you to a more accurate solution.

**Comment by jankogaga** [comment]
Thank you, Nicolas for your answer.

I set which compiler will be used:
In Anaconda (necessary for Windows Python distribution), default compiler is mingw64.
So, I decided not playing with Mingw and switched to C++ compiler in Visual Studio by changing C:\ProgramData\Anaconda3\Lib\distutils\distutils.cfg to

[build]           
compiler=msvc

[build_ext]
compiler=msvc

I downloaded and installed Build Tools for Visual Studio 2017
https://www.visualstudio.com/downloads/#build-tools-for-visual-studio-2017

Then tried installing scikit-surprise and got:

error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual
C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools

I installed then Visual C++ 2015 Build Tools (all features) from provided link.

In the end, I managed to installed scikit-surprise.
